types/block.go

```diff
@@ -226,121 +227,53 @@ func (b *Block) fillHeader() {
 func (b *Block) fillDataAvailabilityHeader() {
 	namespacedShares, dataSharesLen := b.Data.ComputeShares()
 	shares := namespacedShares.RawShares()
+
 	if len(shares) == 0 {
 		// no shares -> no row/colum roots -> hash(empty)
 		b.DataHash = b.DataAvailabilityHeader.Hash()
 		return
 	}
 
+	// create the nmt wrapper to generate row and col commitments
+	squareSize := uint32(math.Sqrt(float64(len(shares))))
+	tree := wrapper.NewErasuredNamespacedMerkleTree(uint64(squareSize))
+
 	// TODO(ismail): for better efficiency and a larger number shares
 	// we should switch to the rsmt2d.LeopardFF16 codec:
-	extendedDataSquare, err := rsmt2d.ComputeExtendedDataSquare(shares, rsmt2d.NewRSGF8Codec(), rsmt2d.NewDefaultTree)
+	extendedDataSquare, err := rsmt2d.ComputeExtendedDataSquare(shares, rsmt2d.NewRSGF8Codec(), tree.Constructor)
```

Comment:
This starts to remind me Java 😬 
`extendedAbstractFactoryForWrappedCodedDataSqaure...`

Reply:
Do you have any concrete suggestion how to make this more go-idiomatic then?

Reply:
Yep, renaming it to `eds`

Reply:
yeah, I'm not the biggest fan of in object data processing either. :man_shrugging: We have two nested layers of abstraction here. I prefer everything in an explicit pipeline. We discussed this specific topic a few times already, and we decided to leave it as is. [ref](https://github.com/lazyledger/rsmt2d/pull/9)

Reply:
> Yep, renaming it to eds

Yeah, I'm not opposed to short variable names if the code is obvious, a few lines long (the smaller the var's scope the shorter the var name could be). Here I'd say it is a matter of taste mostly. I think the name `extendedDataSquare` has little in common with `extendedAbstractFactoryForWrappedCodedDataSqaure`.

> I prefer everything in an explicit pipeline.

Me too but I was also in favour of retiring rsmt2d and make just one concrete instantiation of it (with the only tree we want and everything we need) inside of this repo. For various reasons that party were resolved in the meantime. Let's not get distracted by this right now.